client/src/App.tsx

```diff
@@ -0,0 +1,46 @@
+import './App.css';
+import React, {useEffect, useState} from 'react';
+import RenderReviewList from './components/RenderReviewList';
+import SignInWithSlack from './components/SignInWithSlack';
+import Navbar from './components/Navbar';
+import ChannelFilter from './components/ChannelFilter';
+import {logError} from './services/log';
+import {fetchData} from './services/fetch';
+
+function App() {
+    const queryString = new URLSearchParams(window.location.search);
+    const [status, setStatus] = useState(queryString.get('status') || 'pending');
+    const [user, setUser] = useState(null);
+    const [selectedChannel, setSelectedChannel] = useState(queryString.get('channel') || 'all');
+
+    const handleNavBarClick = (value: string) => {
+        setStatus(value);
+    };
+
+    useEffect(() => {
+        fetchData('/api/user')
+            .then((data) => setUser(data))
+            .catch(logError);
+    }, []);
+
+    return (
+        <div className="App">
+            {!user && (<SignInWithSlack/>)}
```

Comment:
Not a blocker, but it might be good to use ternary operators here since there are times when using `&&` could render the string `0`

> Note that returning a falsy expression will still cause the element after && to be skipped but will return the falsy expression. In the example below, </>0</> will be returned by the render method.

https://legacy.reactjs.org/docs/conditional-rendering.html#inline-if-else-with-conditional-operator

Reply:
Agree.  This is safe because of ! that convert it to boolean expression translate to false which react will not render.  I do agree it can lead to potential problem if not careful.  If the false resolve to false or empty string, then we're fine.  If the condition resolve to 0, yes, it will cause issue.